dbms/src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1470,6 +1471,42 @@ bool ParserFunctionWithKeyValueArguments::parseImpl(Pos & pos, ASTPtr & node, Ex
     return true;
 }
 
+bool ParserTTLElement::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserKeyword s_to_disk("TO DISK");
+    ParserKeyword s_to_volume("TO VOLUME");
+    ParserKeyword s_delete("DELETE");
+    ParserStringLiteral parser_string_literal;
+    ParserExpression parser_exp;
+
+    ASTPtr expr_elem;
+    if (!parser_exp.parse(pos, expr_elem, expected))
+        return false;
+
+    ASTTTLElement::DestinationType destination_type = ASTTTLElement::DestinationType::DELETE;
+    String destination_name;
+    if (s_to_disk.ignore(pos)) {
+        destination_type = ASTTTLElement::DestinationType::DISK;
+    } else if (s_to_volume.ignore(pos)) {
+        destination_type = ASTTTLElement::DestinationType::VOLUME;
+    } else {
+        s_delete.ignore(pos);
+    }
+
+    if (destination_type == ASTTTLElement::DestinationType::DISK || destination_type == ASTTTLElement::DestinationType::VOLUME) {
+        ASTPtr ast_space_name;
+        if (!parser_string_literal.parse(pos, ast_space_name, expected))
+            return false;
+
+        destination_name = ast_space_name->as<ASTLiteral &>().value.get<const String &>();
```

Comment:
the bad cast of there will be a number. Better `tryGet` and apparent exception in case of error.

Reply:
Sorry I can not get the idea, it looks like we do not do that in way more other places.